modules/createTransitionManager.js

```diff
@@ -226,15 +210,6 @@ export default function createTransitionManager(history, routes) {
         if (history.listenBeforeUnload)
           unlistenBeforeUnload = history.listenBeforeUnload(beforeUnloadHook)
       }
-    } else {
-      if (hooks.indexOf(hook) === -1) {
```

Comment:
This is not quite right – `RouteHooks[routeID]` should actually no longer be an array at all, and this `else` branch should instead just overwrite the value.

So this whole thing should be something like:

```
const thereWereNoRouteHooks = !hasAnyProperties(RouteHooks)

const routeID = getRouteID(route)
RouteHooks[routeID] = hook

if (thereWereNoRouteHooks) {
  // ...
```